src/Serilog/Core/LevelOverrideMap.cs

```diff
@@ -68,8 +68,12 @@ public LevelOverride(string context, LoggingLevelSwitch levelSwitch)
     {
         foreach (var levelOverride in _overrides)
         {
-            if (context.StartsWith(levelOverride.Context) &&
-                (context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.'))
+            if (
+                (
+                    context.Length == levelOverride.Context.Length ||
```

Comment:
Sorry for the necroposting on something that's already overkill, but I will forever be pondering whether using a String.Equals into the same length case will be cheaper, or whether `StartsWith` internally special cases this...

Reply:
This is what the current `StartsWith` implementation does (when passing `OrdinalIgnoreCase`):

```csharp
case StringComparison.OrdinalIgnoreCase:
    if (this.Length < value.Length)
    {
        return false;
    }
    return Ordinal.EqualsIgnoreCase(ref this.GetRawStringData(), ref value.GetRawStringData(), value.Length);
```

And this is (effectively, I inlined a method call for brevity) what `Equals` does:

```csharp
case StringComparison.OrdinalIgnoreCase:
    if (a.Length != b.Length)
    {
        return false;
    }

    return Ordinal.EqualsIgnoreCase(ref a.GetRawStringData(), ref b.GetRawStringData(), b.Length);
```

In the case where the length is the same, they both call into the same machinery with the same length, so I think they should be equal here.

Reply:
Thanks Kristian, this has made my day (especially for saving Travis the lookup from an annoying drive by commenter!)